File: parse.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By


def parse_html(driver):
    carrousel = driver.find_element(
        By.XPATH, '//*[@id="_c2yRXMvVOs3N-QazgILgAg93"]/div/div')
    paintings = carrousel.find_elements_by_class_name("klitem")

    array = []
    for p in paintings:
        # get title
        title = p.get_attribute('aria-label')

        # get link
        link = p.get_attribute('href')
        link = link.replace('file://', 'https://www.google.com', 1)

        # get image
        img_box = p.find_element_by_class_name('BA0A6c')
        img = img_box.find_element_by_tag_name('img')
        src = img.get_attribute('src')

        # get date
        try:
            date = p.find_element_by_class_name('klmeta').text
        except:
            date = ''

        result = {
            "name": title,
            "extensions": [date],
            "link": link,
            "image": src
        }

        if date is '':
            result.pop('extensions')

        array.append(result)
    return array


if __name__ == "__main__":
    driver = webdriver.Chrome('/usr/bin/chromedriver')
    
    html_path = os.path.abspath('files/van-gogh-paintings.html')
    driver.get('file://' + html_path)

    array = parse_html(driver)
    print(array)

File: parse.rb
require 'nokogiri'
require 'json'

def parse_carousel(fn)
    html = File.read(fn)

    doc = Nokogiri::HTML(html, nil, 'utf-8')

    first_item = true
    prev_item = nil
    search_data = doc.xpath("//a[@class='klitem']").map do |item|
        name = item.attribute('aria-label').value
        link = "https://www.google.com#{item.attribute('href').value}"
        image_id = item.xpath(".//g-img/img").first.attribute('id').value

        if first_item
            script_regex = /\(function\(\)\{var s='(.*?)';var ii=\['#{image_id}'\];/
            first_item = false
        else
            script_regex = /var ii=\['#{prev_item}'\];_setImagesSrc\(ii,s\);\}\)\(\);\(function\(\)\{var s='(.*?)';var ii=\['#{image_id}'\];/
        end
        prev_item = image_id

        image = script_regex.match(html)&.[](1)&.gsub('\\', '')

        extensions = item.xpath(".//div[@class='ellip klmeta']").map { |i| i.children.map(&:text) }.flatten

        if !extensions.empty?
            next {
                name: name,
                extensions: extensions,
                link: link,
                image: image
            }
        end

        {
            name: name,
            link: link,
            image: image
        }
    end

    {
        artworks: search_data
    }
end